pyccel/ast/low_level_tools.py

```diff
@@ -42,6 +42,39 @@ def iterable_type(self):
         """
         return self._iterable_type
 
+    def __str__(self):
+        return f'Iter[{self._iterable_type}]'
+
+    @property
+    def datatype(self):
+        """
+        The datatype of the object.
+
+        The datatype of the object.
+        """
+        return self
+
+    @property
+    def rank(self):
+        """
+        Number of dimensions of the object.
+
+        Number of dimensions of the object. If the object is a scalar then
+        this is equal to 0.
+        """
+        return 0
```

Comment:
Is this always the case? What if we iterate over a list of lists?

Reply:
The iterator isn't the object inside the container. It is just a container containing a pointer to that object. It still needs unpacking. The only thing you can do with an STC iterator is `elem = iter->ref`. This means that you can never do `iter[i]` so I don't think it needs to be considered as a multi-dimensional object